src/main/java/Duke.java

```diff
@@ -1,10 +1,214 @@
+                DukePrint.printTaskSize(tasks);
+                taskCount++;
+            } catch (StringIndexOutOfBoundsException e) {
+                System.out.println("\uD83D\uDE2D Please enter deadline in the format: 'deadline <name> /by <day> <time>'");
+            }
+            break;
+        case ("event"):
+            // Add an models.Event
+            try {
+                String eventTime = phrase.substring(dividerPosition + 4);
+                phrase = phrase.substring(6, dividerPosition);
+                Event event = new Event(phrase, eventTime);
+                System.out.println("Got it! I've added this task:");
+                tasks.add(event);
+                System.out.println(event);
+                DukePrint.printTaskSize(tasks);
+                taskCount++;
+            } catch (StringIndexOutOfBoundsException e) {
+                System.out.println("\uD83D\uDE2D Please enter event in the format: 'event <name> /at <day> <time>'");
+            }
+            break;
+        case ("delete"):
+            // Deletes a task
+            int toDelete = Integer.parseInt(subStrings[1]) - 1;
+            System.out.println("Noted. I've removed this task:");
+            System.out.println(tasks.get(toDelete).toString());
+            tasks.remove(toDelete);
+            DukePrint.printTaskSize(tasks);
+            break;
+        default:
+            // Invalid models.Task
+            System.out.println("\uD83D\uDE2D Sorry mate I do not understand your request. Please specify task");
+            break;
+        }
+    }
+
+    public static void parseData(String line) {
+        String[] tokens = line.split("#");
+
+        switch (tokens[0]) {
+        case "T":
+            try {
+                boolean isDone = Integer.parseInt(tokens[1]) == 1;
+                Todo todo = new Todo(tokens[2], isDone);
+                tasks.add(todo);
+                taskCount++;
+            } catch (NumberFormatException e) {
+                return;
+            }
+            break;
+        case "D":
+            try {
+                boolean isDone = Integer.parseInt(tokens[1]) == 1;
+                Deadline deadline = new Deadline(tokens[2], tokens[3], isDone);
+                tasks.add(deadline);
+                taskCount++;
+            } catch (NumberFormatException e) {
+                return;
+            }
+            break;
+        case "E":
+            try {
+                boolean isDone = Integer.parseInt(tokens[1]) == 1;
+                Event event = new Event(tokens[2], tokens[3], isDone);
+                tasks.add(event);
+                taskCount++;
+            } catch (NumberFormatException e) {
+                return;
+            }
+            break;
+        default:
+            System.out.println("Invalid data!");
+            break;
+        }
+    }
+
+    public static void saveFile() throws IOException {
+        File path = new File("tasks.txt");
+        if (!path.exists()) {
```

Comment:
Consider merging these into a single boolean condition if possible; it's best to avoid unnecessary nesting